TYPO3.Flow/Classes/TYPO3/Flow/Utility/Ip.php

```diff
@@ -0,0 +1,76 @@
+<?php
+namespace TYPO3\Flow\Utility;
+
+/*
+ * This file is part of the TYPO3.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+/**
+ * A utility class for dealing with IP addresses.
+ *
+ * @Flow\Scope("singleton")
+ */
+class Ip
+{
+    /**
+     * Matches a CIDR range pattern against an IP
+     *
+     * The range can contain IPv4 and IPv6 addresses (including IPv6 wrapped IPv4 addresses).
+     * @see http://tools.ietf.org/html/rfc4632
+     * @see http://tools.ietf.org/html/rfc4291#section-2.3
+     *
+     * Example: 127.0.0.0/24 will match all IP addresses from 127.0.0.0 to 127.0.0.255
+     *          127.0.0.0/31 and 127.0.0.1/31 will both match the IP addresses 127.0.0.0 and 127.0.0.1
+     *          127.0.0.254/31 and 127.0.0.255/31 will both match the IP addresses 127.0.0.254 and 127.0.0.255
+     *          1:2::3:4 will match the IPv6 address written as 1:2:0:0:0:0:3:4 or 1:2::3:4
+     *          ::7F00:1 will match the address written as 127.0.0.1, ::127.0.0.1 or ::7F00:1
+     *          ::1 (IPv6 loopback) will *not* match the address 127.0.0.1
+     *
+     * @param string $ip The IP to match
+     * @param string $range The CIDR range pattern to match against
+     * @return boolean TRUE if the pattern matched, FALSE otherwise
+     */
+    public static function cidrMatch($ip, $range)
```

Comment:
Yeah, that's just a simple refactor of the existing code from the Ip Pattern class into an Utility class, and it's thoroughly tested ( I need to adjust the tests still).

I'm open to replacing it with an external dependency in another PR, though I'm a little cautious with adding external dependencies (that bring their own take of problems) only for singular needs (i.e. this single CIDR matcher).